### PR TITLE
[process-agent] Add tests for check format functions

### PR DESCRIPTION
**Pull request by robertjli**
### What does this PR do?

Add tests for payload format functions in the process check.

### Motivation

Increased test coverage

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] ~Changed code has automated tests for its functionality.~
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] ~If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).~
- [ ] ~If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.~
- [ ] ~If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.~
- [ ] ~If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.~

### CODE REVIEW

File: pkg/process/checks/process.go [hunk 1]
```diff
@@ -390,8 +378,9 @@ func formatCommand(fp *procutil.Process) *model.Command {
 }
 
 func formatIO(fp *procutil.Stats, lastIO *procutil.IOCountersStat, before time.Time) *model.IOStat {
-	// This will be nil for Mac
-	if fp.IOStat == nil {
+	if fp.IORateStat != nil {
+		return formatIORates(fp.IORateStat)
+	} else if fp.IOStat == nil { // This will be nil for Mac
```
Comment (wiyu) [inline pkg/process/checks/process.go:383]: minor, what if we just remove the `else` and have another if statement here as the previous one already returns.

File: pkg/process/checks/process_common_test.go [hunk 2]
```diff
@@ -258,6 +345,69 @@ func TestFormatNetworks(t *testing.T) {
 	}
 }
 
+func TestFormatCPU(t *testing.T) {
+	hostCPUCount = func() int {
+		return 4
+	}
```
Comment (wiyu) [inline pkg/process/checks/process_common_test.go:351]: [minor] Save and reset the previous func that you are overriding incase other tests rely on this.

File: pkg/process/checks/process_nix.go [hunk 3]
```diff
@@ -19,6 +19,10 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/util/system"
 )
 
+var (
+	hostCPUCount = system.HostCPUCount
```
Comment (xornivore) [inline pkg/process/checks/process_nix.go:23]: mind adding a short comment for the intention here (used for tests)

**Review by xornivore — APPROVED** [review]
🙇

File: pkg/process/checks/process_nix.go (hunk 3 above)
Comment (robertjli) [reply pkg/process/checks/process_nix.go:24]: Done!

File: pkg/process/checks/process_common_test.go (hunk 2 above)
Comment (robertjli) [reply pkg/process/checks/process_common_test.go]: Done!

File: pkg/process/checks/process.go (hunk 1 above)
Comment (robertjli) [reply pkg/process/checks/process.go]: Done!

**Comment by robertjli** [comment]
I realized that there's a `formatCPUTimes()` function in both `process_nix.go` and `process_windows.go`, which are used for different build tags. I split the test into separate functions.

**Review by just-chillin — APPROVED** [review]

**Review by wiyu — APPROVED** [review]

**Pull request merged**